src/index.js

```diff
@@ -0,0 +1,73 @@
+const fs = require('fs');
+const { join } = require('path');
+
+class StaticAssetsManager {
+  constructor({
+    staticAssetVersions = {},
+    useStaticAssetVersions = false,
+    staticAssetDirLocal = '/public',
+    staticAssetDirCDN = '/public',
+  }) {
+    this.staticAssetVersions = staticAssetVersions;
+    this.useStaticAssetVersions = useStaticAssetVersions;
+    this.staticAssetDirLocal = staticAssetDirLocal;
+    this.staticAssetDirCDN = staticAssetDirCDN;
+  }
+
+  getStaticUrl({ filename, shouldAppendCacheBust = false }) {
+    const filepath = this.getFilepath({ filename });
+    if (this.useStaticAssetVersions) {
+      const cdnUrl = this.staticAssetVersions[filepath];
+      if (cdnUrl) {
+        return cdnUrl;
+      }
+    }
+    const localUrl = this.getFilepath({ filename, forceLocal: true });
+    return shouldAppendCacheBust
+      ? `${localUrl}?v=${new Date().valueOf()}`
+      : localUrl;
+  }
+
+  setStaticAssetVersions({ staticAssetVersions }) {
+    this.staticAssetVersions = staticAssetVersions;
+    this.useStaticAssetVersions = true;
+  }
+
+  getFilepath({ filename, forceLocal = false }) {
+    return this.useStaticAssetVersions && !forceLocal
+      ? join(this.staticAssetDirCDN, filename)
+      : join(this.staticAssetDirLocal, filename);
+  }
+}
+
+let staticAssetsManager;
+const initializeStaticAssetsManager = ({
+  staticAssetVersionsFilename,
+  staticAssetDirLocal,
+  staticAssetDirCDN,
+}) => {
+  staticAssetsManager = new StaticAssetsManager({ staticAssetDirLocal, staticAssetDirCDN });
+  return new Promise((resolve, reject) => {
+    if (!staticAssetVersionsFilename) {
+      return resolve();
+    }
+
+    return fs.readFile(staticAssetVersionsFilename, 'utf8', (err, contents) => {
+      if (err) {
+        return reject(err);
+      }
+      staticAssetsManager.setStaticAssetVersions({ staticAssetVersions: JSON.parse(contents) });
+      return resolve();
+    });
+  });
+};
+
+const staticUrl = (filename, shouldAppendCacheBust) => staticAssetsManager.getStaticUrl({
```

Comment:
Good one! If we had to pick one pattern, I would use the object/keywords args since it is the one we are using everywhere else. It feels better for maintainability IMO: no param ordering when adding / removing one for instance, or easier to construct params to be passed.

I think I switched to multiple args for `staticUrl` to reduce the "footprint" and have a less verbose function to use: `staticUrl('/buffer.png', true)` instead of `staticUrl({filename: '/buffer.png', cacheBust : true})`.

Happy to go with one or the other :)

Reply:
Thanks @philippemiguet, yeah I was thinking of using one pattern for simplicity sake, but with that thinking, I'm cool with the pattern you already have. We also could support both by checking if the first arg is an object or string, but that could be a future optimization of the library :)